eclipse-collections/src/main/java/org/eclipse/collections/impl/map/sorted/immutable/AbstractImmutableSortedMap.java

```diff
@@ -188,6 +189,9 @@ public void clear()
         return sortedMap.toImmutable();
     }
 
+    @Override
+    public abstract AbstractImmutableSortedSet<K> keySet();
```

Comment:
@johnjstuart I don't think this is needed at all and it constrains things too much IMHO for the subclasses

Reply:
@johnjstuart I'm surprised this happens, worst case scenario, we should expose publicly an interface, not an internal implementation here if possible

Reply:
now it's clear :)